client/src/components/CafeCard.tsx

```diff
@@ -20,71 +19,48 @@ const CafeCard = (props: CardProps) => {
   const ref = useRef<HTMLDivElement>(null);
   const intersection = useIntersection(ref, { threshold: 0.7 });
 
+  const handleScroll = () => {
+    if (ref.current) {
+      const { scrollLeft, clientWidth } = ref.current;
+      const index = Math.round(scrollLeft / clientWidth);
+      setCurrentImageIndex(index);
+    }
+  };
+
   useEffect(() => {
     if (intersection) {
       onIntersect?.(intersection);
     }
   }, [intersection?.isIntersecting]);
 
-  const getImageIndexByOffset = (offset: number) => {
-    const length = cafe.images.length;
-    return (currentImageIndex + length + offset) % length;
-  };
-
-  const getImageByOffset = (offset: number) => cafe.images[getImageIndexByOffset(offset)];
-
-  const handlePrevImage = () => {
-    setCurrentImageIndex(getImageIndexByOffset(-1));
-    announceImageChange('이전 이미지', currentImageIndex - 1);
-  };
-
-  const handleNextImage = () => {
-    setCurrentImageIndex(getImageIndexByOffset(1));
-    announceImageChange('다음 이미지', currentImageIndex + 1);
-  };
-
-  const announceImageChange = (action: string, imageIndex: number) => {
-    const message = `${cafe.name} 카페, ${imageIndex + 1}번째 사진입니다. ${action} 버튼을 클릭하셨습니다.`;
-
-    // 숨겨진 메시지를 화면에 추가하여 스크린 리더가 읽을 수 있게 함
-    // 하드 코딩이라 개선 필요
-
-    const announcementDiv = document.createElement('div');
-    announcementDiv.setAttribute('aria-live', 'polite');
-    announcementDiv.style.position = 'absolute';
-    announcementDiv.style.width = '1px';
-    announcementDiv.style.height = '1px';
-    announcementDiv.style.overflow = 'hidden';
-    document.body.appendChild(announcementDiv);
-
-    announcementDiv.textContent = message;
-
-    setTimeout(() => {
-      announcementDiv.remove();
-    }, 1000);
-  };
+  useEffect(() => {
+    if (ref.current) {
+      ref.current.addEventListener('scroll', handleScroll);
+    }
+    return () => {
+      if (ref.current) {
+        ref.current.removeEventListener('scroll', handleScroll);
+      }
+    };
+  }, []);
```

Comment:
```suggestion
  useEffect(() => {
    ref.current?.addEventListener('scroll', handleScroll);
    return () => {
      ref.current?.removeEventListener('scroll', handleScroll);
    };
  }, []);
```
옵셔널 체이닝을 하면 더 깔끔할 것 같아요!

Reply:
수정하였습니다^^